src/flag_gems/ops/angle.py

```diff
@@ -0,0 +1,57 @@
+import triton
+import triton.language as tl
+import torch
+import math
+from ..utils import pointwise_dynamic,tl_extra_shim
+
+atan2 = tl_extra_shim.atan2
+
+@pointwise_dynamic(
+    is_tensor=[True, True],
+    promotion_methods=[(0, "DEFAULT")]
+)
+@triton.jit
+def angle_func(real, imag):
+    pi = math.pi
+    zero = 0.0
+    one = 1.0
+
+    real_zero_mask = real == zero
+    imag_positive = imag > zero
+
+
+    angle_if_real_zero = tl.where(
+        imag_positive, pi / 2,
+        tl.where(imag < zero, -pi / 2, zero)
+    )
+    real_last=tl.where(real.dtype==torch.float16,real/one,real)
+    imag_last=tl.where(imag.dtype==torch.float16,imag/one,imag)
+    atan_x = atan2(imag_last,real_last)
```

Comment:
tl.extra.libdevice.atan2 seems able to handle the case when real is zero. I think the mask and tl.where statements could be spared.